clients/drcachesim/tools/opcode_mix.h

```diff
@@ -82,8 +83,29 @@ class opcode_mix_t : public analysis_tool_t {
     parallel_shard_error(void *shard_data) override;
 
 protected:
+    struct opcode_category_data_t {
+        opcode_category_data_t()
+            : opcode(0)
+            , category(0)
+        {
+        }
+        opcode_category_data_t(int opcode, uint category)
+            : opcode(opcode)
+            , category(category)
+        {
+        }
+        int opcode;
+        /*
+         * The category field is a uint instead of a dr_instr_category_t because
+         * multiple category bits can be set when an instruction belongs to more
+         * than one category.  We assume 32 bits (i.e., 32 categories) is enough
+         * to be future-proof.
+         */
+        uint category;
+    };
+
     struct worker_data_t {
-        std::unordered_map<app_pc, int> opcode_cache;
+        std::unordered_map<app_pc, opcode_category_data_t> opcode_category_cache;
```

Comment:
Optional: "opcode_data_cache" or "opcode_info_cache" sounds more like it includes several pieces of info and not just a category.  On that same note: "opcode_category_data_t" sounds like just the category and maybe "opcode_data_t" or "opcode_info_t" is more descriptive?  It's subjective though: if you like it how it is better please keep it.

Reply:
I like the "data" variant. Done.